### ISSUE TITLE
ERROR (Query): Unknown asset pair

### ISSUE THREAD

**Issue by jleatham** [Endogen/Telegram-Kraken-Bot#14]
Great app by the way!  Too bad kraken API lag is terrible.

TL;DR: Can I  view the price of other currencies in XBT? e.g., Price of XMR is .015 XBT, etc

In my config file, for 'trade_to_currency, everything worked with USD (I think, I haven't made an actual trade), but I could list the USD price of each coin.  However, when I change the 'trade_to_currency' to XBT, it looks like trading is working(haven't tested) but I cannot list prices in XBT.  Instead, it gives me the error: ERROR (Query): Unknown asset pair .  I'll try and take some time to look at the API to see what is going on but figured I would ask y'all and see if this is a known issue or maybe I am missing something.

Other feedback, I had never used telegram but the instructions were straight forward, thanks!  The only thing that wasn't immediately clear was that the user_id for telegram is the actual digit ID, and not the user_name.

**Comment by Endogen** [comment]
Thanks for reporting. I will take a look at it soon. Sorry that this is taking some time...

**Comment by Endogen** [comment]
So, now i have the time to look into that :)

> Great app by the way! Too bad kraken API lag is terrible.

Thanks! Yeah, Kraken can't handle the load - it's a shame!

> TL;DR: Can I view the price of other currencies in XBT? e.g., Price of XMR is .015 XBT, etc

Actually, i never thought about that! I will try to get that working but right now i don't think it really works. If it kind of works, then that's a coincidence. :) The `trade_to_currency` parameter was only intended for "real" currencies like `EUR` or `USD`.

user_id for telegram: Thanks for the hint! Will update the README to make that more clear

**Comment by jleatham** [comment]
I guess while we are at this, the monero wallets sometimes need a payment ID/note along with the wallet ID.  I was hoping to help with this more but there are other exchanges that are so much better than kraken that I've kind of given up on them at the moment.

**Comment by Endogen** [comment]
Yeah that is really an issue. Which exchange do you like the most? I'm thinking about supporting other ones.

**Comment by Endogen** [comment]
> I guess while we are at this, the monero wallets sometimes need a payment ID/note along with the wallet ID.

Do you mean that there needs to be an option to enter the necessary ID / note in order to withdraw coins? So currently funding doesn't work for Monero?

**Comment by hertzdog** [comment]
Hi Endogen, great project!
I am having a similar issue. My trade_to_currency is EUR: when asking the value for EOS, I got the "Uknown asset pair" - because Kraken has not a EOS/EUR, but only EOS/ETC and EOS/XBT. 

Unfortunately I am not a python programmer (neither a programmer at all), but I was thinking if something like the following makes sense:
`if ALTCRYPTO/ trade_to_currency is not defined: find ALTCRYPTO/CRYPTO pairs;
foreach pair check if CRYPTO/trade_to_currency is defined. 
Define ALTCRYPTO/ trade_to_currency as the average of ALTCRYPTO/CRYPTO * CRYPTO/trade_to_currency
Return the value of ALTCRYPTO/ trade_to_currency`

**Comment by Endogen** [comment]
Hey @hertzdog, your lucky! I'm currently working on that feature and i almost got it finished. :) I guess i need a few days till it's finished. Stay tuned!

**Comment by Endogen** [comment]
It will work like this: You will have to define for each coin that you want to trade, which `trade_to_currency` you want to use.

I guess that's just the first step, since it would be nice to have the possibility to trade every possible pair. But at the moment, i'm happy getting it to the mentioned state.

**Comment by Endogen** [comment]
I just updated the master branch to a new version of the bot that supports trading every asset-pair available on kraken
